File: persiandatepicker/src/main/java/ir/hamsaa/persiandatepicker/date/PersianDateImpl.java
package ir.hamsaa.persiandatepicker.date;

import java.util.Date;

import ir.hamsaa.persiandatepicker.api.PersianPickerDate;

public class PersianDateImpl implements PersianPickerDate {

    private PersianDateFixedLeapYear persianDate;

    public PersianDateImpl() {
        persianDate = new PersianDateFixedLeapYear();
    }

    @Override
    public void setDate(Long timestamp) {
        persianDate = new PersianDateFixedLeapYear(timestamp);
    }

    @Override
    public void setDate(Date date) {
        persianDate = new PersianDateFixedLeapYear(date);
    }

    @Override
    public void setDate(int persianYear, int persianMonth, int persianDay) {
        try {
            persianDate.setShYear(persianYear);
            persianDate.setShMonth(persianMonth);
            persianDate.setShDay(persianDay);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    @Override
    public int getPersianYear() {
        return persianDate.getShYear();
    }

    @Override
    public int getPersianMonth() {
        return persianDate.getShMonth();
    }

    @Override
    public int getPersianDay() {
        return persianDate.getShDay();
    }

    @Override
    public int getGregorianYear() {
        return persianDate.getGrgYear();
    }

    @Override
    public int getGregorianMonth() {
        return persianDate.getGrgMonth();
    }

    @Override
    public int getGregorianDay() {
        return persianDate.getGrgDay();
    }

    @Override
    public int getDayOfWeek() {
        return persianDate.dayOfWeek();
    }

    @Override
    public String getPersianMonthName() {
        return persianDate.monthName();
    }

    @Override
    public String getPersianDayOfWeekName() {
        return persianDate.dayName();
    }

    @Override
    public String getPersianLongDate() {
        return getPersianDayOfWeekName() + "  " + getPersianDay() + "  " + getPersianMonthName() + "  " + getPersianYear();
    }

    @Override
    public Date getGregorianDate() {
        return persianDate.toDate();
    }

    @Override
    public long getTimestamp() {
        return persianDate.getTime();
    }

    @Override
    public boolean isLeapYear() {
        return persianDate.isLeap();
    }

    public static boolean isLeapYear(int year) {
        return new PersianDateFixedLeapYear().isLeap(year);
    }
}

File: persiandatepicker/src/main/java/ir/hamsaa/persiandatepicker/date/PersianDateImpl.kt
package ir.hamsaa.persiandatepicker.date

import ir.hamsaa.persiandatepicker.api.PersianPickerDate
import saman.zamani.persiandate.PersianDate
import java.util.*

class PersianDateImpl : PersianPickerDate {

    private var persianDate: PersianDate

    init {
        persianDate = PersianDate()
    }

    override fun setDate(timestamp: Long) {
        persianDate = PersianDate(timestamp)
    }

    override fun setDate(date: Date) {
        persianDate = PersianDate(date)
    }

    override fun setDate(persianYear: Int, persianMonth: Int, persianDay: Int) {
        persianDate.shYear = persianYear;
        persianDate.shMonth = persianMonth;
        persianDate.shDay = persianDay;
    }

    override fun getPersianYear(): Int {
        return persianDate.shYear
    }

    override fun getPersianMonth(): Int {
        return persianDate.shMonth
    }

    override fun getPersianDay(): Int {
        return persianDate.shDay
    }

    override fun getGregorianYear(): Int {
        return persianDate.grgYear
    }

    override fun getGregorianMonth(): Int {
        return persianDate.grgMonth
    }

    override fun getGregorianDay(): Int {
        return persianDate.grgDay
    }

    override fun getDayOfWeek(): Int {
        return persianDate.dayOfWeek()
    }

    override fun getPersianMonthName(): String {
        return persianDate.monthName()
    }

    override fun getPersianDayOfWeekName(): String {
        return persianDate.dayName()
    }

    override fun getPersianLongDate(): String {
        return getPersianDayOfWeekName() + "  " + getPersianDay() + "  " + getPersianMonthName() + "  " + getPersianYear()
    }

    override fun getGregorianDate(): Date {
        return persianDate.toDate()
    }

    override fun getTimestamp(): Long {
        return persianDate.time
    }

}